utils/sync/limiter_test.go

```diff
@@ -0,0 +1,152 @@
+package sync_test
+
+import (
+	"context"
+	"strconv"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/rudderlabs/rudder-server/services/stats/memstats"
+	miscsync "github.com/rudderlabs/rudder-server/utils/sync"
+	"github.com/stretchr/testify/require"
+)
+
+func TestLimiter(t *testing.T) {
+	t.Run("without priority", func(t *testing.T) {
+		ctx, cancel := context.WithCancel(context.Background())
+		var wg sync.WaitGroup
+		ms := memstats.New()
+
+		statsTriggerCh := make(chan time.Time)
+		triggerFn := func() <-chan time.Time {
+			return statsTriggerCh
+		}
+
+		limiter := miscsync.NewLimiter(ctx, &wg, "test", 1, ms, miscsync.WithLimiterStatsTriggerFunc(triggerFn))
+		var counter int
+		statsTriggerCh <- time.Now()
+		statsTriggerCh <- time.Now()
```

Comment:
How come we need to trigger twice?

Reply:
To make sure that the first operation that was triggered has been completed

Reply:
To make sure that the first operation that was triggered has been completed